packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -630,69 +637,111 @@ class SrvImplDart implements Srv<SocketConnector> {
     return socketConnector;
   }
 
-  Future _handleMultiConnectRequest(
+  Future<void> _handleMultiConnectRequest(
+    SocketConnector sc,
+    InternetAddress relayAddress,
+    DataTransformer? encrypter,
+    DataTransformer? decrypter,
+  ) async {
+    logger.info('_runDaemonSideMulti'
+        ' Control socket received connect request - '
+        ' creating new socketToSocket connection');
+
+    InternetAddress localAddress = await resolveRequestedLocalHost();
+
```

Comment:
In the code below, we're taking direct control of creating the socket connections (to the relay and to the requested local `host:port`) rather than calling `SocketConnector.socketToSocket`.

This is required because (1) the order really matters here (connect to relay first, connect to local `host:port` second) (2) write and flush the auth string to the relay before trying to create the local socket (3) ensure that there are no uncaught exceptions on either socket